app/src/main/java/com/globant/example/mentorapp/home/presentation/view/fragment/ListUsersViewFragment.java

```diff
@@ -0,0 +1,144 @@
+ */
+public class ListUsersViewFragment extends LifecycleFragment implements BaseView {
+
+    public static final String LIST_TAG = "listUserFragment";
+    @Inject
+    public ListUsersPresenterImpl presenter;
+    protected StaggeredGridLayoutManager staggeredGridLayoutManager;
+    private SharedUserViewModel model;
+    private ListUsersAdapter listUsersAdapter;
+    private RecyclerView listUsersRecyclerView;
+    private BaseActivity parent;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((MentorApplication) getActivity().getApplicationContext()).getApplicationComponent().inject(this);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_home_screen, container, false);
+        model = ViewModelProviders.of(this).get(SharedUserViewModel.class);
+        parent = (BaseActivity) getActivity();
+        staggeredGridLayoutManager = new StaggeredGridLayoutManager(
+                getResources().getInteger(R.integer.number_of_users_columns), LinearLayoutManager.VERTICAL);
+        if (model.getUsers() != null || model.getUsers().getValue() != null) {
+            listUsersAdapter = new ListUsersAdapter(model.getUsers().getValue(), getContext());
+            parent.hideProgress();
+        }
+        listUsersRecyclerView = (RecyclerView) view.findViewById(R.id.list_item_container);
+        defineRecyclerView(listUsersRecyclerView, listUsersAdapter);
+        return view;
+    }
+
+    @Override
+    public void render(BaseModel baseModel) {
+        ListUsersViewModel usersModel = (ListUsersViewModel) baseModel;
+        if (usersModel.getUsers() != null) {
+            model.setUsers(usersModel.getUsers());
+            usersReady();
+        } else if (usersModel.getError() != null)
+            switch (usersModel.getError()) {
+                case ERROR_CONNECTION:
+                    usersErrorConnectivity();
+                    break;
+                case ERROR_RESPONSE:
+                    usersErrorHttp();
+                    break;
+            }
+        if (usersModel.getProgress() != null) {
+            if (usersModel.getProgress()) {
+                parent.showProgress();
+            } else {
+                parent.hideProgress();
+            }
+        }
+    }
+
+    private void usersReady() {
+        listUsersAdapter = new ListUsersAdapter(model.getUsers().getValue(), getContext());
+        listUsersRecyclerView.setAdapter(listUsersAdapter);
+        listUsersAdapter.notifyDataSetChanged();
+    }
+
+    private void usersErrorHttp() {
+        snackBarMessage(R.string.http_error_message);
+    }
+
+    private void usersErrorConnectivity() {
+        snackBarMessage(R.string.communication_error_message);
+    }
+
+    private void snackBarMessage(int stringResource) {
```

Comment:
You can execute this in *usersErrorHttp* and *usersErrorConnectivity* since it's just one line